api/catalog/api/views/image_views.py

```diff
@@ -132,7 +132,12 @@ def watermark(self, request, *_, **__):
         params.is_valid(raise_exception=True)
 
         image = self.get_object()
-        image_url = image.url
+
+        # Media providers may provide large images in ``image.url``,
+        # causing the services to slow down or time out. So, we will use
+        # ``image.thumbnail`` (if available) for thumbnail generation.
+        image_url = image.thumbnail if image.thumbnail else image.url
```

Comment:
This change applies to the `watermark` method, but the original issue was related to thumbnail generation. Can you apply the same change to line 113 in the `thumbnail` method and remove the change from this line in the `watermark` method? For watermark generation I believe we would want the original image (and in any case, is out of scope of the original issue).

Reply:
Hey @sarayourfriend, Thank you so much for the review, I thought since the line was mentioned in the issue i had to change it here, I'll make the required changes.

Reply:
The issue linked to the line in the `thumbnail` method, but the line is identical between the two methods, so it is an easy mix-up to make. Thanks for correcting it so quickly.